lab_5_scrapper/scrapper.py

```diff
@@ -127,11 +235,29 @@ def _extract_url(self, article_bs: BeautifulSoup) -> str:
         Returns:
             str: Url from HTML
         """
+        link = article_bs.find('a', class_="post-item-link")
+        if link:
+            return self.url_pattern + link.get('href')
+        return ''
```

Comment:
I tried not to use a loop to extract the url but it extracts only one link and writes texts only from it. Where am I wrong, here or in find_articles?

Reply:
you need to call _extract_url method several times in a loop from find_articles method 

each time you call _extract_url, you get one link and check it

Reply:
Fixed